userProfile/privateProfile.mjs

```diff
@@ -29,29 +30,56 @@ router.use(
   })
 )
 
-router.get("/profile", async (req, res) => {
-  const user = await req.user 
-  if (!user) return respondWithError(res, 401, "Unauthorized user") 
+router.get("/profile", auth0Middleware(), async (req, res) => {
+  const user = await req.user
+  if (!user) return respondWithError(res, 401, "Unauthorized user")
+  
   const userObj = new User(user._id)
-  const userProfile = await userObj.getSelf()
-  res.set("Content-Type", "application/json; charset=utf-8") 
-
-  res.status(200).json(userProfile)
+  userObj.getSelf()
+  .then(userData => {  
+    res.status(200).json(userData);
+  })
+  .catch(error => { 
+    res.status(error.status || error.code || 500).json({
+      error: error.message || 'An error occurred while fetching the user data.',
+      status: error.status || 'Error'
+    });
+  })
 })
 
-router.get("/projects", async (req, res) => {
-  const {_id} = await req.user 
+router.get("/test/profile", auth0Middleware(),  (req, res) => { 
+  const userId = req.user
+ 
+  const user = new User(userId); 
+  user.getUserById()
+    .then(userData => {  
+      res.json(userData);
+    })
+    .catch(error => {
+      console.log("error caught here")
+      // Handle the error, e.g., send an error response
+      res.status(error.status || 500).json({
+        error: error.message || 'An error occurred while fetching the user data.',
+        status: error.status || 'Error'
+      }); 
+    })
+  })
 
-  if (!_id) return respondWithError(res, 401, "Unauthorized user")
+router.get("/projects", auth0Middleware(), async (req, res) => {
+  const user = await req.user
```

Comment:
You need to update this route to do the error throughput like you did with "/profile".

![image](https://github.com/CenterForDigitalHumanities/TPEN-services/assets/3287006/e1fae4be-b27a-4ce8-b180-9fa8e0398675)

Reply:
I'm unable to replicate this. The error comes out fine on my end
<img width="555" alt="image" src="https://github.com/CenterForDigitalHumanities/TPEN-services/assets/111019083/f2335546-9b1f-48c4-8ee2-58bcb3851f1e">


for the following error

<img width="302" alt="image" src="https://github.com/CenterForDigitalHumanities/TPEN-services/assets/111019083/572034e7-05aa-492f-a1b1-dbd476f549a7">